music/views.py

```diff
@@ -1,6 +1,26 @@
 from django.shortcuts import render
+from django.db.models import Q
+
 from .models import Song
 
 def index(request):
-    allSongs = Song.objects.all()
-    return render(request, template_name="index.html", context={"allSongs" : allSongs})
+    allSongs = Song.objects.all().order_by('-last_updated')
+    return render(request, template_name="music/index.html", context={"allSongs" : allSongs})
+
+
+def search_songs(request): 
+    template_path = 'music/search_result.html'
+    
+    search_query = request.GET.get('search', None)
+
+    if search_query: 
+        search_result = Song.objects.filter(
+            Q(songName__icontains=search_query) | 
+            Q(album__albumName__icontains=search_query) | 
+            Q(album__artist__artistName__icontains=search_query)
+        )
+    else: 
+        search_result = Song.objects.all()
+        
+    context = {'search_result' : search_result, 'search_query' : search_query}
```

Comment:
This code is okay, but in larger application, mostly devs keeps parameter within required method, like here in render method. everything necessary will go in it. you kept here for readable purpose its good.

Reply:
yes, I try to make code readable. If something goes wrong, I jump on to the variable if it needs any changes, and I see this cleaner. But, this is my perspective, if industry and senior devs put all the necessary things inside the method, then I will follow this too.

Reply:
every other review is done, I am waiting for this one. If it needs to serve everything in the method, then I will change this, but if you say it's okey, then I will I'm good to go for another review.